src/index.js

```diff
@@ -0,0 +1,20 @@
+const fs = require('fs');
+const path = require('path');
+
+function createAsyncapiFile(generator) {
+  const asyncapi = generator.originalAsyncAPI;
+  let extension;
+  
+  try {
+    JSON.parse(asyncapi);
+    extension = 'json';
+  } catch (e) {
+    extension = 'yaml';
+  }
+
+  fs.writeFileSync(path.resolve(generator.targetDir, `asyncapi.${extension}`), asyncapi);
```

Comment:
There is an issue, please have a look on the same hook in [java-spring template](https://github.com/asyncapi/java-spring-template/blob/96cedfee25a74eaf1f614f1dcaec2916f5d5a23f/hooks/addApiFile.js#L16)
For java it's make sense to place it in specified directory of a project.

Reply:
@Tenischev @fmvilas 

What if I make this hook look like:
```
const fs = require('fs');
const path = require('path');
const xfs = require('fs.extra');

function createAsyncapiFile1(generator) {
  const asyncapi = generator.originalAsyncAPI;
  const targetDir = generator.targetDir;
  const customDirInTarget = generator.templateParams.asyncapiFileDir;
  
  let extension;
  
  try {
    JSON.parse(asyncapi);
    extension = 'json';
  } catch (e) {
    extension = 'yaml';
  }
  
  const getCustomFileLocation = (target, dir, filename) => {
    console.log(dir, "dir")
    xfs.mkdirpSync(path.resolve(target, dir)) 
    return path.resolve(target, dir, filename) 
  }

  const outputFileName = `asyncapi.${extension}`;
  console.log( customDirInTarget)
  
  console.log(path.resolve(targetDir, customDirInTarget, outputFileName))
  const asyncapiOutputLocation = customDirInTarget 
    ? getCustomFileLocation(targetDir, customDirInTarget, outputFileName)
    : path.resolve(targetDir, outputFileName);

  fs.writeFileSync(asyncapiOutputLocation, asyncapi);
}

module.exports = {
  'generate:after': createAsyncapiFile1
};
```

And in Java template in config it would be enough to add such param:
```
    "asyncapiFileDir": {
      "description": "Custom location of the AsyncAPI file that you provided as an input in generation. By default it is located in the root of the output directory"
    }
```

and document that you can generate Java service with template using `-p asyncapiFileDir=src/main/resources/api`

Reply:
@derberg I'm not so sure... It's very good to allow customizing of target dir, but i do not see reason to ask user to define path for async api, especially when he/she is not familiar with template.

Reply:
@Tenischev `asyncapiFileDir` is not marked as mandatory here and by default, the file goes into the root, so the user is not forced to use the flag unless he wants to have it in `src/main/resources/api`

Reply:
well, more specifically, as the author of the template, I want to be able to set the default path

Reply:
@Tenischev sounds fair, but this should not be part of hooks functionality in my opinion, but the config itself. We should just introduce `parameters[param].default`. So you can reuse hooks that have parameters, and when adding them to your template config, you can specify a default value for the param. How does it sound?

Reply:
Great idea! Also will simplify some code in java-spring template.

Reply:
@Tenischev I created an issue https://github.com/asyncapi/generator/issues/339

Now, regarding java template, I will now jump on it, working on PR to support new hooks. What do you want me to do with the hook about creating asyncapi file, just keep it there and only refactor, or use the hook I proposed here and make `asyncapiFileDir` mandatory param in your template? until #339 is done

Reply:
I think for now leave as it is, and after #339 make an update of template